gymnasdicts/utils.py

```diff
@@ -57,4 +57,46 @@ def parse_pointer(pointer_str: str) -> Tuple[str, ...]:
         # it really is implemented as a raw Exception in this package
         raise ValueError(f"{exception}")
 
-    return _pointer_to_tuple(pointer)
+    nested_tuples = _pointer_to_tuple(pointer)
+    if not flatten:
+        return nested_tuples
+    return sum(nested_tuples, tuple())
+
+
+def aggregate_two_items(left: Any, right: Any, path: Dict) -> Any:
+    """
+    :example:
+        >>> a = {"a": {"x": [3], "y": 5, "z": 9}}
+        >>> b = {"a": {"x": [4], "y": 5, "z": 10}}
+        >>> aggregate_two_items(a, b, {"a": ["x", "z"]})
+        {'a': {'x': [3, 4], 'y': 5, 'z': 19}}
+
+    """
+
+    assert type(left) == type(right)
+
+    if isinstance(left, list):  # apply to each item
+        assert len(left) == len(right)
+        return [
+            aggregate_two_items(left_item, right_item, path)
+            for left_item, right_item in zip(left, right)
+        ]
+
+    if isinstance(left, dict):
+        assert left.keys() == right.keys()
+        assert set(path).issubset(left), "path contains fields not in payload!"
+
+        ret = {}
+        for key in left:
+            if key not in path:  # check equal and move on
+                ret[key] = aggregate_two_items(left[key], right[key], {})
+            elif isinstance(path, (list, tuple)):  # end of the line, sum these objects
```

Comment:
Would you always want to sum them? What about the case where it is string or indeed two different types e.g. str and int? Would it make more sense to always create a list of the original values?

Reply:
> Would it make more sense to always create a list of the original values?
i think this is a good idea. i will take a look

Reply:
@ldomic cf7281e harder than i realised! i am now wondering if it wouldn't be better if it always collected the terms into a dictionary rather than list... what do you think?

Reply:
I think that looks much better as now you could load this result in something like Pandas and get the relevant value of the cost or rainfall instead of the indiscriminate sum! Nice!